modules/nw-ovn-kubernetes-migration-about.adoc

```diff
@@ -0,0 +1,103 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.adoc
+
+[id="nw-ovn-kubernetes-migration-about_{context}"]
+= Migration to the OVN-Kubernetes network provider
+
+Migrating to the OVN-Kubernetes Container Network Interface (CNI) default network provider is a manual process that includes some downtime during which your cluster is unreachable. Although a rollback procedure is provided, the migration is intended to be a one way process.
+
+[NOTE]
+====
+A migration to the OVN-Kubernetes network provider is supported on clusters installed on only bare metal hardware.
+====
+
+[id="how-the-migration-process-works_{context}"]
+== How the migration process works
```

Comment:
@pliurh, is there a high level overview of how the migration actually works conceptually?

Thanks!

Reply:
Users annotate the network.operator so that the CNO can allow changes to the cluster network. After changing the defaultNetwork type in network.config, the CNO will destroy the openshift-sdn objects from the cluster then deploy ovn-kubernetes ones. It will also re-render the multus objects to work with openshift-ovn-kubernetes. We don't want MCO to interrupt at this phase, so we suspend it temporarily. After openshift-sdn objects removed, multus pods recreated, and ovn-kubernetes objects created, the new cluster network is ready for pods to attach to. However, all the existing pods are still attached to openshift-sdn network. So we trigger the manual reboot for all the nodes. After this reboot, all the pods will attach to ovn-kubernetes. But this is an intermediate stage, where nodes haven't been reconfigured for ovn-kubernetes shared gateway mode. So we resume all MachineConfig Pool, so that MCO can apply the newly rendered MachineConfig for all the nodes. After MCO does its job, the migration is done. Users can check whether the openshift pods and their workload pods are running as expected.